src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,39 @@
+import classes from "./Modal.module.css";
+import ReactDom from "react-dom";
+
+interface ModalProps {
+  show: boolean;
+  close: (params?: any) => any;
+  title: string;
+  children?: React.ReactNode;
+}
+
+const Modal: React.FC<ModalProps> = ({ show, close, title, children }) => {
+  return ReactDom.createPortal(
+    <>
+      {show ? (
+        <div onClick={() => close()} className={classes.modalContainer}>
+          <div onClick={(e) => e.stopPropagation()} className={classes.modalBody}>
+            <div className={classes.modalHeader}>
+              <div className={classes.modalTitle}>{title}</div>
+              <button onClick={() => close()} className={classes.modalClose}>
+                &times;
+              </button>
+            </div>
+            <div className={classes.modalContent}>{children}</div>
+
+            <div className={classes.modalFooter}>
+              <button onClick={() => close()} className={classes.modalCancel}>
+                Cancel
+              </button>
```

Comment:
in JSX you do not have to create an arrow f-n each time. You can simply wright:
```suggestion
              <button onClick={close} className={classes.modalCancel}>
                Cancel
              </button>
```

Reply:
Sınce I am not confident yet about state and lifecycle events in react, I asked one of my friends and he said, the states are async so it would be good to use pass an arrow function this kind of situation (like modal). Thats why. But you are right It seems unnecessary to me as well.